samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c

```diff
@@ -83,25 +85,35 @@ static bool await_connection_result(k_timeout_t timeout)
 static void ready_timeout_work_fn(struct k_work *work)
 {
 	ARG_UNUSED(work);
-	LOG_INF("nRF Cloud connection did not become ready in time, disconnecting and retrying...");
-	disconnect_cloud();
+	if (!atomic_get(&initial_association) || !IS_ENABLED(CONFIG_NRF_CLOUD_MQTT)) {
+		LOG_INF("nRF Cloud connection did not become ready in time, "
+			"disconnecting and retrying...");
+		disconnect_cloud();
+	} else {
+		int err;
+
+		LOG_INF("Checking for association");
+		err = nrf_cloud_check_association();
```

Comment:
Rather than implement this at the application level, I personally think we should use a workqueue to implement it at the library level. From application perspective, we just want to receive the associated notification when association completes -- If periodically calling nct_ping during JITP is required for that, I feel we should implement that.


From MSS perspective, having this association check here is, IMO, fairly confusing. Here we are in the timeout which gets called if association did not occur within the time expected by the sample (the ready timeout). So this line appears at first glance to be a totally separate mechanism to check for that association a different way

In reality, it is just a means to make sure that the nrf_cloud lib ***eventually*** fires the event it was supposed to fire, but again IMO we should not drive this at the application level